src/server/chatController.py

```diff
@@ -145,11 +145,18 @@ def __init__(self, *args, **kw):
 
   def opened(self):
     self.username = cherrypy.session['username']
+    if self.username in self.ChatClients:
+      cherrypy.log("WARNING: User {0} already had a ChatWS. Replacing".format(self.username))
+
     self.ChatClients[self.username] = self
     cherrypy.log("User {0} ({1}) ChatWS connected".format(self.username, self.peer_address))
 
   def closed(self, code, reason=None):
-    del self.ChatClients[self.username]
+    try:
+      del self.ChatClients[self.username]
+    except:
+      cherrypy.log("ERROR: ChatWS for {0} was not in dict.".format(self.username))
```

Comment:
The try catch could be replace by [dict.pop](https://docs.python.org/2/library/stdtypes.html#dict.pop) ... but we could lose the log ...